pol-core/pol/multi/house.cpp

```diff
@@ -419,7 +448,19 @@ namespace Pol {
 		Items::Item* item = Core::find_toplevel_item( tmp_serial );
 		if ( item != NULL )
 		{
-		  components_.push_back( Component( item ) );
+          if( ! add_component(Component(item)) ) {
+            std::ostringstream os;
+            os << "Couldn't add component " << item->serial << " to house " << this->serial << "." << std::endl;
+            UHouse* contHouse = item->house();
+            if( contHouse == nullptr ) {
+              os << "This is probably a core bug. Please report it on the forums.";
+            } else {
+              os << "This item is already part of house " << contHouse->serial << "." << std::endl;
+              os << "Allowing an item to be a component in two different houses was a bug," << std::endl;
+              os << "please also fix your save data.";
+            }
+            throw std::runtime_error(os.str());
+          }

```

Comment:
Instead of std::ostringstream fmt::Writer would be better. In addition the serials are currently not shown in hex. ".. 0x" << fmt::hexu(serial) would do it.